collection-scripts/version

```diff
@@ -1,16 +1,17 @@
 #!/usr/bin/env bash
 
 function version() {
-  # get version from image
-  version=$( \
-    oc status | grep '^pod' | \
-    sed -n -r -e 's/.*([[:digit:]]+\.[[:digit:]]+(:?\.[[:digit:]])?(:?-[^@]+)?).*/\1/p' \
-  )
 
-  # if version not found, fallback to imageID
-  [ -z "${version}" ] && version=$(oc status | grep '^pod.*runs' | sed -r -e 's/^pod.*runs //')
+  # get gather container image and imageID
+  read -r image imageID < <(oc get pod ${POD_NAME} -o jsonpath='{range .status.containerStatuses[?(@.name=="gather")]}{.image} {.imageID}{end}')
```

Comment:
Assuming the script is not run outside a container, it's ok to read `POD_NAME` variable ([always set in the must-gather pod spec](https://github.com/openshift/oc/blob/8e20cf6e8a3c85335e6caf4a9ae53dcc8fe2e9fa/pkg/cli/admin/mustgather/mustgather.go#L748)). Nevertheless:
- reading `POD_NAME` env creates a new dependency on the `pkg/cli/admin/mustgather` code (when the `POD_NAME` variable is removed, the version will fall back into "Unknown")
- `.status.containerStatuses` does not always have to be set at the time when the `version` command is invoked

What about to update the building pipeline to inject the must-gather version directly? So `version()` can be a simple reading of a file?

Reply:
@sanchezl I am a bit indifferent to the solution here. What do you think is best/easiest?

Reply:
Looks like I have the following available (with example values from 4.12.8):
```
"BUILD_RELEASE=202210242028.p0.ge105f12.assembly.stream",                                                                     
"BUILD_VERSION=v4.12.0",                                                                                                      
"OS_GIT_MAJOR=4",                                                                                                             
"OS_GIT_MINOR=12",                                                                                                            
"OS_GIT_PATCH=0",                                                                                                             
"OS_GIT_TREE_STATE=clean",                                                                                                    
"OS_GIT_VERSION=4.12.0-202210242028.p0.ge105f12.assembly.stream-e105f12",                                                     
"SOURCE_GIT_TREE_STATE=clean",                                                                                                
"OS_GIT_COMMIT=e105f12
```
Looks like `OS_GIT_VERSION` might be appropriate.

Reply:
@sanchezl , is it possible with that `OS_GIT_VERSION` to know the exact version/commit of the must-gather scripts (https://github.com/openshift/must-gather/commits/master/collection-scripts) used? Maybe `OS_GIT_COMMIT`, or is it for a different repository?